### PR TITLE
iss018  - addition of imperial measurements

### PR DESCRIPTION
**Pull request by DnH2016**
Added new sections to horusmapper.cfg.example
Added to config.py to read in new settings - conversion factors and unit selection
Modified tables.js and balloons.js
Tested on live telemetry data.

### CODE REVIEW

File: horusmapper.cfg.example [hunk 1]
```diff
@@ -200,3 +211,30 @@ bearing_color = black
...
+
+# unitselection allows choice of metric - the default or imperial1 - horizontal miles and feet for short distances, horizontal miles per hour,  vertical feet, vertical feet per minute
+
+unitselection = metric
+# unitselection = imperial1
+
+
+# Imperial conversion factors
+
+m_to_ft = 3.28084
```
Comment (darksidelemm) [inline horusmapper.cfg.example]: I don't think these conversions need to be in the configuration file. They can be hardcoded somewhere. 
Remove the conversions, just leave the unitselection and switch_miles_feet parameters.

File: static/js/balloon.js [hunk 2]
```diff
@@ -151,6 +153,54 @@ function updateSummaryDisplay(){
...
+function updateSummaryDisplayImperial1(){
+    
+    // Update the 'Payload Summary' display.
+    var _summary_update = {id:1};
+    // See if there is any payload data.
+    if (balloon_positions.hasOwnProperty(balloon_currently_following) == true){
+        // There is balloon data!
+        var _latest_telem = balloon_positions[balloon_currently_following].latest_data;
+        
+        _summary_update.alt = (_latest_telem.position[2]*chase_config['m_to_ft']).toFixed(0) + "ft (" + (_latest_telem.max_alt*chase_config['m_to_ft']).toFixed(0) + "ft)";
```
Comment (darksidelemm) [inline static/js/balloon.js]: I don't see any need to refer out to config parameters for what are essentially constants. These values are not going to change, so may as well hardcode them. Same applies in numerous other places.

File: horusmapper.cfg.example (hunk 1 above)
Comment (DnH2016) [reply horusmapper.cfg.example]: Removed and left unit_selection and switch_miles_feet

File: static/js/balloon.js (hunk 2 above)
Comment (DnH2016) [reply static/js/balloon.js]: Hardcoded there and numerous other places.

**Comment by darksidelemm** [comment]
Running this up at the moment, looking pretty good! 
I would probably suggest changing the name in the config from 'imperial1' to 'imperial' unless you have intentions of adding an imperial2?

**Comment by darksidelemm** [comment]
Also maybe make a note in the config that this affects the status displays only - not the settings (e.g. burst altitude, range rings). Probably gets a bit messy to convert all that to imperial and back again too (the mapping system, and the predictor all operate in metric)

**Comment by DnH2016** [comment]
There are many versions of imperial, unfortunately. Feet / miles, yards / miles, knots; that's why I put the number on. What do you want to do? 

I forgot the range rings, sorry, but I chose not do the predictor stuff. Since the algorithm runs in metric, it somehow felt wrong. I had misgivings about doing a conversion on the telemetry box, that should just be the raw data. I would like to add units into the burst altitude and descent rate box legends - is that OK with you?

**Comment by darksidelemm** [comment]
I think let's just limit it at the status displays for now, that should be sufficient. Leave it as the one unit set that you have now.

I guess the only other place where it might be useful to change units is on the PolyLine Measure widget: https://github.com/projecthorus/chasemapper/blob/master/templates/index.html#L251
It looks like its just an argument to set the units for that.

**Comment by DnH2016** [comment]
Ah - completely missed that. I'll turn that round now, whilst there is a balloon in the air. I think you want me to leave imperial1 as imperial1, but I don't intend to add any more units - iss018 asked for miles feet and feet/min.

**Comment by darksidelemm** [comment]
Nah, change imperial1 to imperial. If people want other options they can add them another time.

**Comment by DnH2016** [comment]
All changes done, but needs testing on some live data.

**Comment by DnH2016** [comment]
Tested on S2510499

**Pull request merged**